# 捕获异常

try:

​	尝试执行的代码

except:

​	处理



错误类型捕获

try:

​	...

except 错误类型1:

​	...

except (错误类型2, ...):

​	...

except Exception as result:

​	未知错误

else:

​	没异常执行

finally:

​	始终执行





主动抛出异常



如果希望抛出异常

创建一个Exception 类

用raise抛出异常



ex = Exception("fail")

raise